multi-int-store/src/main/java/com/connexta/multiintstore/controllers/StorageController.java

```diff
@@ -36,16 +41,20 @@ public ResponseEntity store(
       @PathVariable String ingestId, @RequestBody(required = false) JsonNode body) {
     LOGGER.info("Received callback for ingestId {}", ingestId);
 
-    Object callback = validator.parse(body);
-
-    if (callback != null) {
+    try {
+      final Object callback = validator.parse(body);
       manager.handleGeneralCallback(callback, ingestId);
-    } else {
+    } catch (UnsupportedCallbackException e) {
+      LOGGER.info("Returning Bad Request", e);
       return new ResponseEntity(HttpStatus.BAD_REQUEST);
+    } catch (ClientException | ServerException e) {
+      LOGGER.info("Returning Internal Error due to retrieval", e);
+      return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
+    } catch (StorageException | DuplicateIdException e) {
+      LOGGER.info("Returning Internal Error due to storage", e);
+      return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
✏️ I know that the implementation of this will probably changes, but it's weird to me that we're catching exceptions here that are only thrown for metadata callbacks. I think at least we could update the log messages to say that the problem was with the metadata.